deploy

```diff
@@ -3,684 +3,618 @@
+        },
+        body: {
+            key_id: config.keyId,
+            block_id: config.blockId,
+            id: config.handlers.before,
+            code: config.signalCode,
+            channels: 'opengrowth.signals',
+            event: 'js-before-publish',
+            type: 'js',
+            log_level: 'debug',
+            name: 'Signal Handler',
+            output: 'opengrowth.signals-info'
         }
-    }).catch( error => {
-        console.error( 'error', error );
-    } );
+    }).catch(error => {
+        console.error('Failed to upload signal handler'.black.bgRed);
+        console.error(error.black.bgRed);
+    });
 }
 
 // =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
-// Create a block
+// Upload Delight/After Handler Code
 // =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
-function create_app(name) {
+function uploadDelightHandler() {
     return http({
-        method  : 'POST'
-    ,   json    : true
-    ,   uri     : 'https://admin.pubnub.com/api/apps'
-    ,   headers : { 'X-Session-Token' : config.login.token }
-    ,   body    : {
-            owner_id : config.owner_id
-        ,   name: name
+        method: 'PUT',
+        json: true,
+        uri: 'https://admin.pubnub.com/api/v1' +
+            `/blocks/key/${config.keyId}/` +
+            `event_handler/${config.handlers.after}`,
+        headers: {
+            'X-Session-Token': config.token
+        },
+        body: {
+            key_id: config.keyId,
+            block_id: config.blockId,
+            id: config.handlers.after,
+            code: config.delightCode,
+            channels: 'opengrowth.signals',
+            event: 'js-after-publish',
+            type: 'js',
+            log_level: 'debug',
+            name: 'Delight Handler',
+            output: 'opengrowth.delights-info'
         }
-    }).catch( error => {
-        console.error( 'error', error );
-    } );
+    }).catch(error => {
+        console.error('Failed to upload delight handler'.black.bgRed);
+        console.error(error.black.bgRed);
+    });
 }
 
 // =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
-// Create a keyset
+// Get Event Handler ids
 // =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
-function create_keyset(app_id, name) {
-    return http({
-        method  : 'POST'
-    ,   json    : true
-    ,   uri     : 'https://admin.pubnub.com/api/keys'
-    ,   headers : { 'X-Session-Token' : config.login.token }
-    ,   body    : { 
-            app_id: app_id
-        ,   properties : { name : name }
-        }
-    }).catch( error => {
-        console.error( 'error', error );
-    } );
+async function getEventHandlers() {
+    let list = await listEventHandlers(config.keyId, config.blockId);
+    list = list.payload[0].event_handlers;
+    let payload = {
+        'before': null,
+        'after': null
+    }
+    for (let i in list) {
```

Comment:
Here you can take advantage of `for ... of` instead of `for ... in`. With
`for (let handler of list)`
the references would look like
`handler.event`
instead of
`list[i].event`